integration/querydsl/blaze-persistence-querydsl-testsuite/src/test/java/com/blazebit/persistence/querydsl/BasicQueryTest.java

```diff
@@ -0,0 +1,744 @@
+                    .with(idHolderCTE, idHolderCTE.id).as(new BlazeJPAQuery<Document>(entityManager, cbf)
+                            .fromValues(document, Collections.singleton(theBook))
+                            .select(document.id))
+                    .select(idHolderCTE.id).from(idHolderCTE)
+                    .fetch();
+
+            assertNotNull(fetch);
+        });
+    }
+
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class, NoMySQLOld.class, NoDB2.class })
+    public void testRecursiveCTEUnion() {
+        Assume.assumeTrue(dbmsDialect.supportsWithClause());
+
+        doInJPA(entityManager -> {
+            List<Long> fetch = new BlazeJPAQuery<Document>(entityManager, cbf)
+                    .withRecursive(idHolderCTE, idHolderCTE.id).as(unionAll(select(document.id).from(document).where(document.id.eq(1L)), select(document.id).from(document)
+                            .join(idHolderCTE).on(idHolderCTE.id.add(1L).eq(document.id))))
+                    .select(idHolderCTE.id).from(idHolderCTE)
+                    .fetch();
+
+            assertNotNull(fetch);
+        });
+    }
+
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class, NoMySQLOld.class })
+    public void testRecursiveBindBuilder() {
+        Assume.assumeTrue(dbmsDialect.supportsWithClause());
+
+        doInJPA(entityManager -> {
+            List<Long> fetch = new BlazeJPAQuery<Document>(entityManager, cbf)
+                    .withRecursive(idHolderCTE, idHolderCTE.id)
+                        .as(new BlazeJPAQuery<>().unionAll(
+                                select(document.id).from(document).where(document.id.eq(1L)),
+                                select(document.id).from(document).join(idHolderCTE).on(idHolderCTE.id.add(1L).eq(document.id))))
+                    .select(idHolderCTE.id).from(idHolderCTE)
+                    .fetch();
+
+            assertNotNull(fetch);
+        });
+    }
+
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class })
+    public void testInlineEntityWithLimit() {
+        doInJPA(entityManager -> {
+            QRecursiveEntity recursiveEntity = new QRecursiveEntity("t");
+
+            List<RecursiveEntity> fetch = new BlazeJPAQuery<RecursiveEntity>(entityManager, cbf)
+                    .select(recursiveEntity)
+                    .from(select(recursiveEntity)
+                            .from(recursiveEntity)
+                            .where(recursiveEntity.parent.name.eq("root1"))
+                            .orderBy(recursiveEntity.name.asc())
+                            .limit(1L), recursiveEntity)
+                    .fetch();
+
+            assertNotNull(fetch);
+        });
+    }
+
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class, NoHibernate42.class, NoHibernate43.class, NoHibernate50.class, NoHibernate51.class})
```

Comment:
Add a comment like e.g. `// NOTE: No advanced sql support for datanucleus, eclipselink and openjpa yet`

Apart from that, why do you exclude old Hibernate versions? Inner entity joins are emulated in older versions.